packages/batch-submitter/src/exec/run-batch-submitter.ts

```diff
@@ -105,6 +115,66 @@ export const run = async () => {
     logger = new Logger({ name })
   }
 
+  const useHardhat = config.bool('use-hardhat', !!env.USE_HARDAT)
```

Comment:
```suggestion
  const useHardhat = config.bool('use-hardhat', !!env.USE_HARDHAT)
```
is the double negative here to get the variable to be `true` even if undefined? i think an `||` would be more readable here for that

Reply:
Using the double negative syntax is shorthand for a typecast to a boolean. A single `!` type casts it to the negation and the second `!` cancels out the negation and returns it to the original truthiness value. It is a relatively common pattern in JS, a replacement for using `Boolean(...)` to type cast. I don't really see anybody ever use `Boolean(...)`, probably because its more verbose?

> is the double negative here to get the variable to be true even if undefined?
No its not, that would be a single negation (undefined is falsey so a single `!` would make it truthy). We only wan't this mode to operate when the env var `USE_HARDHAT` is defined